### PR TITLE
pkg/sockops: use native go API for mount

### PR DESCRIPTION
**Pull request by nirmoy**
Replace exec.Command(mount) with syscall.Mount

Signed-off-by: Nirmoy Das <ndas@suse.de>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6283)

### CODE REVIEW

File: pkg/sockops/sockops.go [hunk 1]
```diff
@@ -85,11 +85,10 @@ func mountCgroup() error {
 		return fmt.Errorf("%s is a file which is not a directory", cgroupRoot)
 	}
 
-	mntArgs := []string{"-t", "cgroup2", "none", cgroupRoot}
-	_, err = exec.Command(prog, mntArgs...).CombinedOutput()
-	if err != nil {
-		return fmt.Errorf("Failed to mount %s: %s", cgroupRoot, err)
+	if err := syscall.Mount(cgroupRoot, cgroupRoot, "cgroup2", 0, ""); err != nil {
```
Comment (vadorovsky) [inline pkg/sockops/sockops.go]: Please use `mountinfo.FilesystemTypeCgroup2` const (https://github.com/cilium/cilium/blob/master/pkg/mountinfo/mountinfo.go#L29)

**Review by vadorovsky — APPROVED** [review]
One minor comment, but LGTM overall. We should check more more often whether Cilium is not overusing `exec`.

**Comment by vadorovsky** [comment]
test-me-please

**Comment by vadorovsky** [comment]
test-me-please

File: pkg/sockops/sockops.go [hunk 2]
```diff
@@ -85,11 +85,10 @@ func mountCgroup() error {
 		return fmt.Errorf("%s is a file which is not a directory", cgroupRoot)
 	}
 
-	mntArgs := []string{"-t", "cgroup2", "none", cgroupRoot}
-	_, err = exec.Command(prog, mntArgs...).CombinedOutput()
-	if err != nil {
-		return fmt.Errorf("Failed to mount %s: %s", cgroupRoot, err)
+	if err := syscall.Mount(cgroupRoot, cgroupRoot, mountinfo.FilesystemTypeCgroup2, 0, ""); err != nil {
```
Comment (aanm) [inline pkg/sockops/sockops.go]: @nirmoy should this be `syscall.Mount("none"`...?

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by aanm — APPROVED** [review]

**Comment by nirmoy** [comment]
test-me-please

**Review by vadorovsky — APPROVED** [review]

**Comment by vadorovsky** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by tgraf** [comment]
Nice!

**Pull request merged**